File: app/src/main/java/com/example/pengalatdite/aplikasisederhana/screen/newslist/NewsListViewMvc.java
package com.example.pengalatdite.aplikasisederhana.screen.newslist;

import android.view.View;

import com.example.pengalatdite.aplikasisederhana.data.DataNews;

import java.util.List;

public interface NewsListViewMvc {
    View getRootView();

    void setDataNews(List<DataNews> dataNews);
}

File: app/src/main/java/com/example/pengalatdite/aplikasisederhana/screen/newslist/NewsListViewMvc.kt
package com.example.pengalatdite.aplikasisederhana.screen.newslist

import android.view.View
import com.example.pengalatdite.aplikasisederhana.data.DataNews
import com.example.pengalatdite.aplikasisederhana.screen.newslist.NewsListViewMvcImpl.Listener

interface NewsListViewMvc {
    fun getRootView(): View
    fun bindData(list: List<DataNews>)
    fun registerListener(listener: Listener)
    fun unregisterListener(listener: Listener)
}